pandas/core/arrays/base.py

```diff
@@ -53,6 +53,13 @@ class ExtensionArray(object):
     * factorize / _values_for_factorize
     * argsort / _values_for_argsort
 
+    For logical operators, the default is to return a Series of boolean.
+    However, if the underlying ExtensionDtype overrides the logical
+    operators, then the implementer may want to have an ExtensionArray
+    subclass contain the result.  This can be done by changing the property
+    _logical_result from its default value of None to the _from_sequence
```

Comment:
Why is it needed to have this property? Can't we simply detect whether the result is a boolean numpy array or again an ExtensionArray ?

Reply:
I can't speak for the author, but my assumption was that this has to do with some of the spaghetti-code in `ops._bool_method_SERIES`, where sometimes a bool-dtype is returned and other times an int-dtype is returned (and datetimelike are currently all broken, see #19972, #19759).  Straightening out this mess independently of EA implementations is part of the plan referred to above.

Reply:
Actually, in my use case, I need the boolean operators to return an object that represents the relation.  I'm using pandas on top of 2 different libraries (that functionally are the same) where the operators (x <= y), (x >= y) and (x == y) are not booleans, but objects representing the relations.